app/controllers/concerns/search.rb

```diff
@@ -18,6 +18,23 @@ def remove_prefix_if_necessary(query)
     query.strip
   end
 
+  # Returns an ActiveRecord::Relation of publishers for pagination
+  def publishers_search(publishers, query)
+    search_query = remove_prefix_if_necessary(params[:q])
+
+    # Simple optimization to only search for the things that the admins search for the most
+    if is_email?(search_query)
+      results = publishers.where('email ILIKE ?', "%#{search_query}%")
+    elsif is_promo_code?(search_query)
+      results = publishers.left_joins(:channels).joins(channels: :promo_registration).where(promo_registrations: { referral_code: search_query.upcase })
+    else
+      search_query = "%#{search_query}%" unless is_a_uuid?(search_query)
```

Comment:
Would you move `is_a_uuid?` from `app/controllers/admin/publishers_controller.rb` to this file? Otherwise this looks great